### ISSUE TITLE
no match for argument: dracut-kiwi-oem-dump and dracut-kiwi-oem-repart

### ISSUE THREAD

**Issue by talk2drys** [OSInside/kiwi#2380]
## Problem description
Just confused  and would appreciate it if anyone could explain, i decided to rebuild centos7 just for testing every other packages built except for `dracut-kiwi-oem-dump` and `dracut-kiwi-oem-repart`, no provider for those package, i could not find any repositories with those 2 packages  for centos 7 but rpm was avalable for 8 and above. so my question is how is centos7 built.

## Description File
I am using the same description file from the kiwi-description project no modification still got same error

```xml
<?xml version="1.0" encoding="utf-8"?>

<image schemaversion="6.8" name="CentOS-07.0_appliance">
    <description type="system">
        <author>Marcus Schaefer</author>
        <contact>ms@suse.de</contact>
        <specification>CentOS Enterprise 7 Appliance</specification>
    </description>
    <profiles>
        <profile name="Live" description="Live image" import="true"/>
        <profile name="Virtual" description="Simple Disk image"/>
        <profile name="Disk" description="Expandable Disk image"/>
    </profiles>
    <strip type="delete">
        <file name="/usr/lib*/python*"/>
        <file name="/usr/lib*/gconv"/>
        <file name="/usr/share/zoneinfo"/>
        <file name="/usr/share/i18n"/>
        <file name="/lib/i686/nosegneg"/>
        <file name="/lib/kbd"/>
    </strip>
    <preferences>
        <version>1.2.0</version>
        <packagemanager>yum</packagemanager>
        <bootsplash-theme>charge</bootsplash-theme>
        <locale>en_US</locale>
        <keytable>us</keytable>
        <timezone>UTC</timezone>
        <rpm-check-signatures>false</rpm-check-signatures>
    </preferences>
    <preferences profiles="Live">
        <type image="iso" flags="overlay" hybrid="true" firmware="uefi" hybridpersistent_filesystem="ext4" hybridpersistent="true"/>
    </preferences>
    <preferences profiles="Virtual">
        <type image="vmx" primary="true" filesystem="ext3" kernelcmdline="rhgb" bootloader="grub2" firmware="bios" editbootinstall="linuxefi_patch.sh"/>
    </preferences>
    <preferences profiles="Disk">
        <type image="oem" initrd_system="dracut" filesystem="ext3" installiso="true" bootloader="grub2" firmware="uefi">
            <oemconfig>
                <oem-systemsize>2048</oem-systemsize>
                <oem-swap>true</oem-swap>
                <oem-swapsize>200</oem-swapsize>
                <oem-multipath-scan>false</oem-multipath-scan>
            </oemconfig>
        </type>
    </preferences>
    <users>
        <user password="$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0" home="/root" name="root" groups="root"/>
    </users>
    <repository type="rpm-md" alias="Fedora-EPEL7">
        <source path="http://download.fedoraproject.org/pub/epel/7/x86_64"/>
    </repository>
    <repository type="rpm-md" alias="CentOS7-Extras">
        <source path="http://mirror.centos.org/centos-7/7/extras/x86_64"/>
    </repository>
    <repository type="rpm-md" alias="CentOS7-Updates">
        <source path="http://mirror.centos.org/centos-7/7/updates/x86_64"/>
    </repository>
    <repository type="rpm-md" alias="CentOS7">
        <source path="http://mirror.centos.org/centos-7/7/os/x86_64"/>
    </repository>
    <packages type="image">
        <namedCollection name="core"/>
        <namedCollection name="console-internet"/>
        <package name="syslinux"/>
        <package name="grub2"/>
        <package name="kernel"/>
        <package name="plymouth-theme-charge"/>
    </packages>
    <packages type="iso">
        <package name="dracut-kiwi-live"/>
    </packages>
    <packages type="oem" profiles="Disk">
        <package name="dracut-kiwi-oem-repart"/>
        <package name="dracut-kiwi-oem-dump"/>
    </packages>
    <packages type="bootstrap">
        <package name="filesystem"/>
        <package name="basesystem"/>
        <package name="yum-plugin-priorities"/>
        <package name="nextgen-yum4"/>
        <package name="grub2-efi-x64-modules"/>
        <package name="grub2-efi-x64"/>
        <package name="shim-x64" arch="x86_64"/>
    </packages>
</image>
```

# Error
```
appliance-provisioner-kiwi-ng-1  | [ INFO    ]: 20:24:37 | --> collection: core
appliance-provisioner-kiwi-ng-1  | [ DEBUG   ]: 20:24:37 | EXEC: [chroot /build/abuja/disk/build/image-root dnf --config /kiwi_o7rczi97 -y --releasever=0 --setopt=install_weak_deps=False install dracut-kiwi-oem-dump dracut-ki
wi-oem-repart grub2 kernel plymouth-theme-charge syslinux @console-internet @core]
[ INFO    ]: Processing: [                                        ] 0%[ DEBUG   ]: 20:24:51 | system: Last metadata expiration check: 0:00:40 ago on Tue Oct 10 20:24:03 2023.
appliance-provisioner-kiwi-ng-1  | [ DEBUG   ]: 20:24:51 | system: No match for argument: dracut-kiwi-oem-dump
appliance-provisioner-kiwi-ng-1  | [ DEBUG   ]: 20:24:51 | system: No match for argument: dracut-kiwi-oem-repart
[ INFO    ]: Processing: [########################################] 100%
appliance-provisioner-kiwi-ng-1  | [ ERROR   ]: 20:24:51 | KiwiInstallPhaseFailed: System package installation failed: Failed to set locale, defaulting to C
appliance-provisioner-kiwi-ng-1  | Error: Unable to find a match: dracut-kiwi-oem-dump dracut-kiwi-oem-repart
appliance-provisioner-kiwi-ng-1  |
appliance-provisioner-kiwi-ng-1  | [ INFO    ]: 20:24:51 | Cleaning up SystemPrepare instance
appliance-provisioner-kiwi-ng-1  | [ DEBUG   ]: 20:24:51 | EXEC: [mountpoint -q /build/abuja/disk/build/image-root/var/cache/kiwi]
```

## Steps to reproduce the behaviour
just build the above description file

## OS and Software information

* KIWI version:   `KIWI (next generation) version 9.24.61`
* Operating system host version:  `Opensuse Tumbleweed  build 20230917`
* Operating system target version:  `Centos 7`

**Comment by schaefi** [comment]
Hmm, it's probably a question for @Conan-Kudo as I think he did the packaging for CentOS.

So the source of these dracut modules belongs to kiwi and the packaging that we do includes the mentioned sub-packages, but we no longer build packages for CentOS7 in our integration tests, only v8 and v9.

In the "generic" spec you can find the packaging definitions:

* https://github.com/OSInside/kiwi/blob/master/package/python-kiwi-spec-template#L496

In the worst case you might want to rebuild the package for centos7

Not sure if this helps

**Comment by Conan-Kudo** [comment]
I don't ship it in EPEL 7.

**Comment by Conan-Kudo** [comment]
For a more complete answer: we dropped support for RHEL 7 when we dropped support for Python 2 some years ago. It's a lot of work to support a Python 3 application on RHEL 7, and with RHEL 7 being so close to its end of life, we decided to no longer attempt it.

Even back when we did support it, the CentOS 7 description was already wrong, since it was missing the Appliance Builder repo required to get the dracut modules. That's our bad, and we'll clean that up.

The KIWI project follows the recommendation from the CentOS project to consider upgrading to CentOS Stream 8 or CentOS Stream 9, as both are currently supported by the KIWI project and have builds available.
